vspace/src/test/java/edu/asu/diging/vspace/core/services/impl/SpaceManagerTest.java

```diff
@@ -116,23 +125,32 @@ public void test_storeSpace_success() throws FileStorageException {
         Mockito.verify(spaceDisplayRepo).save((SpaceDisplay)spaceDisplay);
     }
 
-
-
-    @Test(expected = SpaceDoesNotExistException.class)
-    public void test_deleteSpaceById_whenIdIsNull() throws SpaceDoesNotExistException {
-        Mockito.doThrow(IllegalArgumentException.class)
-        .when(spaceRepo).deleteById(null);
+    @Test
+    public void test_deleteSpaceById_whenIdIsNull() throws SpaceDoesNotExistException{
+        Mockito.when(managerToTest.getSpace(spaceId)).thenReturn(null).thenReturn(null);
+        Assert.assertEquals(null, managerToTest.getSpace(null));
         managerToTest.deleteSpaceById(null);
+        Assert.assertEquals(null, managerToTest.getSpace(null));
     }
 
-    @Test(expected = SpaceDoesNotExistException.class)
-    public void test_deleteSpaceById_forNonExistentId() throws SpaceDoesNotExistException {
-        String imgId = "imgId";
-        Mockito.doThrow(EmptyResultDataAccessException.class)
-        .when(spaceRepo).deleteById(imgId);
-        managerToTest.deleteSpaceById(imgId);
+    @Test
+    public void test_deleteSpaceById_forNonExistentId() {  
+        Mockito.when(managerToTest.getSpace(spaceId)).thenReturn(null).thenReturn(null);
```

Comment:
why are there two `thenReturn(null)` calls?

Reply:
2 thenReturns are to mimic one call to the repo before the deletion and one after the delete method is called to show the difference.

Reply:
For a nonexistant Id, there will be no difference as nothing gets deleted.

Reply:
see my other comment